### PR TITLE
Be more lenient with try/except in _showwarning

### PR DESCRIPTION
**Pull request by bozdoz**
Was getting an IndexError from the `formatWarning` function.  Alternatively, you could alter that function to check for values before trying to get a value from an index.

### CODE REVIEW

**Comment by bozdoz** [comment]
Related to #67

**Comment by moser** [comment]
@bozdoz: May I suggest to use
`except Exception:`
instead the bare exception? :-)
see https://moserei.de/2017/09/20/python-bare-except.html

**Comment by bozdoz** [comment]
@moser done

File: PyPDF2/pdf.py [hunk 1]
```diff
@@ -1157,12 +1157,8 @@ def _showwarning(message, category, filename, lineno, file=warndest, line=None):
                 if file is None:
                     file = sys.stderr
                 try:
-                    # It is possible for sys.stderr to be defined as None, most commonly in the case that the script
-                    # is being run vida pythonw.exe on Windows. In this case, just swallow the warning.
-                    # See also https://docs.python.org/3/library/sys.html#sys.__stderr__
-                    if file is not None:
-                        file.write(formatWarning(message, category, filename, lineno, line))
-                except IOError:
+                    file.write(formatWarning(message, category, filename, lineno, line))
```
Comment (MartinThoma) [inline PyPDF2/pdf.py:1160]:
```suggestion
                    # It is possible for sys.stderr to be defined as None, most commonly in the case that the script
                    # is being run vida pythonw.exe on Windows. In this case, just swallow the warning.
                    # See also https://docs.python.org/3/library/sys.html#sys.__stderr__
                    if file is not None:
                        file.write(formatWarning(message, category, filename, lineno, line))
```

**Comment by MartinThoma** [comment]
That issue was fixed via https://github.com/py-pdf/PyPDF2/pull/612/files

**Pull request closed without merging**